components/match_table/commons/match_table.lua

```diff
@@ -0,0 +1,753 @@
+---@field timeIsExact boolean
+---@field timeZone string
+---@field liquipediatier string?
+---@field liquipediatiertype string?
+---@field displayName string
+---@field icon string?
+---@field iconDark string?
+---@field pageName string
+---@field vods {index: number, link: string}[]
+---@field type string
+---@field result MatchTableMatchResult
+---@field game string?
+
+---@class MatchTableMatchResult
+---@field opponent match2opponent
+---@field vs match2opponent
+---@field winner number
+---@field resultType string?
+---@field countGames boolean
+
+---@class MatchTable
+---@operator call(table): MatchTable
+---@field args table
+---@field title Title
+---@field config MatchTableConfig
+---@field matches MatchTableMatch[]
+---@field stats {matches: {w: number, d: number, l: number}, games: {w: number, d: number, l: number}}
+---@field display Html
+local MatchTable = Class.new(function(self, args)
+	self.args = args or {}
+	self.title = mw.title.getCurrentTitle()
+end)
+
+---@return self
+function MatchTable:readConfig()
+	local args = self.args
+
+	local mode = args.tableMode
+	assert(mode == PLAYER_MODE or mode == TEAM_MODE, 'Unsupported "|tableMode=" input')
+
+	local opponents = self:_readOpponents(mode)
+
+	self.config = {
+		mode = mode,
+		limit = tonumber(args.limit),
+		displayGameIcons = Logic.readBool(args.gameIcons),
+		showResult = Logic.nilOr(Logic.readBoolOrNil(args.showResult), true),
+		timeRange = self:readTimeRange(),
+		aliases = self:readAliases(mode),
+		vs = {},
+		title = args.title,
+		showTier = not Logic.readBool(args.hide_tier),
+		showIcon = not Logic.readBool(args.hide_icon),
+		showVod = Logic.readBool(args.vod),
+		showStats = Logic.nilOr(Logic.readBoolOrNil(args.stats), true),
+		showOpponent = Logic.nilOr(Logic.readBoolOrNil(args.showOpponent), #opponents > 1 or mode == PLAYER_MODE),
+	}
+
+	Array.forEach(opponents, function(opponent)
+		Table.mergeInto(self.config.aliases, self:getOpponentAliases(mode, opponent))
+	end)
+
+	local vsMode = args.vsMode or mode
+	assert(vsMode == PLAYER_MODE or vsMode == TEAM_MODE, 'Unsupported "|vsMode=" input')
+
+	Array.forEach(self:_readVsOpponents(mode), function(opponent)
+		Table.mergeInto(self.config.vs, self:getOpponentAliases(mode, opponent))
+	end)
+
+	return self
+end
+
+---@param mode MatchTableMode
+---@return standardOpponent[]
+function MatchTable:_readOpponents(mode)
+	local base = mode == PLAYER_MODE and 'player' or 'team'
+	local inputs = self:_readOpponentInputsFromBase(base)
+
+	if Logic.isEmpty(inputs) then
+		assert(self.title.namespace == 0, 'Required ' .. base .. '= argument')
+		table.insert(inputs, self.title.rootText)
+	end
+
+	return Array.map(inputs, function(input) return self:_readOpponent(mode, input) end)
+end
+
+---@param mode MatchTableMode
+---@return standardOpponent[]
+function MatchTable:_readVsOpponents(mode)
+	local inputs = self:_readOpponentInputsFromBase('vs' .. (mode == PLAYER_MODE and 'player' or 'team'))
```

Comment:
Could also make the input mode independent?
```suggestion
	local inputs = self:_readOpponentInputsFromBase('vsOpponent')
```

Reply:
would break the old usage